lua/lint/linters/cppcheck.lua

```diff
@@ -1,25 +1,31 @@
 -- cppcheck <= 1.84 doesn't support {column} so the start_col group is ambiguous
 local pattern = [[([^:]*):(%d*):([^:]*): %[([^%]\]*)%] ([^:]*): (.*)]]
-local groups = { 'file', 'lnum', 'col', 'code', 'severity', 'message' }
+local groups = { "file", "lnum", "col", "code", "severity", "message" }
 local severity_map = {
-  ['error'] = vim.diagnostic.severity.ERROR,
-  ['warning'] = vim.diagnostic.severity.WARN,
-  ['performance'] = vim.diagnostic.severity.WARN,
-  ['style'] = vim.diagnostic.severity.INFO,
-  ['information'] = vim.diagnostic.severity.INFO,
+  ["error"] = vim.diagnostic.severity.ERROR,
+  ["warning"] = vim.diagnostic.severity.WARN,
+  ["performance"] = vim.diagnostic.severity.WARN,
+  ["style"] = vim.diagnostic.severity.INFO,
+  ["information"] = vim.diagnostic.severity.INFO,
 }
 
 return {
-  cmd = 'cppcheck',
+  cmd = "cppcheck",
   stdin = false,
   args = {
-    '--enable=warning,style,performance,information',
-    '--language=c++',
-    '--inline-suppr',
-    '--quiet',
-    '--cppcheck-build-dir=build',
-    '--template={file}:{line}:{column}: [{id}] {severity}: {message}',
+    "--enable=warning,style,performance,information",
+    (function()
+      if vim.bo.filetype == "cpp" then
+        return "--language=c++"
+      else
+        return "--language=c"
+      end
+    end)(),
```

Comment:
This should stay a function to ensure it's lazy evaluated on use and not on import. Otherwise if you open first a cpp project and then a c project it would be frozen to c++.

Reply:
Right, sorry, didn't know that. Fixed.